metricbeat/module/mongodb/collstats/collstats.go

```diff
@@ -57,39 +54,28 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch methods implements the data gathering and data conversion to the right
 // format. It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
-func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 	// instantiate direct connections to each of the configured Mongo hosts
 	mongoSession, err := mongodb.NewDirectSession(m.DialInfo)
 	if err != nil {
-		logger.Error(err)
-		reporter.Error(err)
```

Comment:
This is still missing here 😄 same in line 76.

Reply:
Ah, missed one!

Reply:
Wait, do we need to add anything there? My understanding is that we only need to add the error reporting back in where we're not actually returning, i.e, inside a loop

Reply:
My understanding is `logger.Error(err)` is to log the error and `reporter.Error(err)` is to actually report an event with error msg. So if we only have a `return errors.Wrap(...)` here, then we are missing the report an event part.

Reply:
If we return an error, it's not only logged but also reported: https://github.com/elastic/beats/blob/master/metricbeat/mb/module/wrapper.go#L243